AEPMessaging/Tests/UnitTests/MessagingTests.swift

```diff
@@ -1079,6 +1083,114 @@ class MessagingTests: XCTestCase {
+        // setup
+        let NEW_PUSH_TOKEN = "different push token"
+        messagingProperties.pushIdentifier = MOCK_PUSH_TOKEN
+        messaging = Messaging(runtime: mockRuntime, rulesEngine: mockMessagingRulesEngine, contentCardRulesEngine: mockContentCardRulesEngine, expectedSurfaceUri: mockSurface.uri, cache: mockCache, messagingProperties: messagingProperties)
+        messaging.onRegistered()
+        let mockConfig = [EXPERIENCE_CLOUD_ORG: MOCK_EXP_ORG_ID, MessagingConstants.SharedState.Configuration.PUSH_FORCE_SYNC: false] as [String : Any]
+
+        let eventData: [String: Any] = [MessagingConstants.Event.Data.Key.PUSH_IDENTIFIER: NEW_PUSH_TOKEN]
+
+        let event = Event(name: "handleProcessEvent", type: EventType.genericIdentity, source: EventSource.requestContent, data: eventData)
+        mockRuntime.simulateSharedState(for: MessagingConstants.SharedState.Configuration.NAME, data: (value: mockConfig, status: SharedStateStatus.set))
+        mockRuntime.simulateXDMSharedState(for: MessagingConstants.SharedState.EdgeIdentity.NAME, data: (value: SampleEdgeIdentityState, status: SharedStateStatus.set))
+
+        // test
+        XCTAssertNoThrow(messaging.handleProcessEvent(event))
+        XCTAssertNotNil(mockRuntime.dispatchedEvents)
+        let pushTokenEvent = mockRuntime.firstEvent
+        XCTAssertEqual(EventType.edge, pushTokenEvent?.type)
+        XCTAssertEqual(EventSource.requestContent, pushTokenEvent?.source)
+        XCTAssertEqual(NEW_PUSH_TOKEN, getDispatchedEventPushToken(event: pushTokenEvent))
+    }
+    
+    func testPushTokenSync_whenNoExistingPushToken() {
+        // setup
+        let NEW_PUSH_TOKEN = "new push token"
+        let mockConfig = [EXPERIENCE_CLOUD_ORG: MOCK_EXP_ORG_ID, MessagingConstants.SharedState.Configuration.PUSH_FORCE_SYNC: false] as [String : Any]
+
+        let eventData: [String: Any] = [MessagingConstants.Event.Data.Key.PUSH_IDENTIFIER: NEW_PUSH_TOKEN]
+
+        let event = Event(name: "handleProcessEvent", type: EventType.genericIdentity, source: EventSource.requestContent, data: eventData)
+        mockRuntime.simulateSharedState(for: MessagingConstants.SharedState.Configuration.NAME, data: (value: mockConfig, status: SharedStateStatus.set))
+        mockRuntime.simulateXDMSharedState(for: MessagingConstants.SharedState.EdgeIdentity.NAME, data: (value: SampleEdgeIdentityState, status: SharedStateStatus.set))
+
+        // test
+        XCTAssertNoThrow(messaging.handleProcessEvent(event))
+        XCTAssertNotNil(mockRuntime.dispatchedEvents)
+        let pushTokenEvent = mockRuntime.firstEvent
+        XCTAssertEqual(EventType.edge, pushTokenEvent?.type)
+        XCTAssertEqual(EventSource.requestContent, pushTokenEvent?.source)
+        XCTAssertEqual(NEW_PUSH_TOKEN, getDispatchedEventPushToken(event: pushTokenEvent))
+    }
+    
+    func testMultiplePushTokenSync_whenTokenMatchesAndWithinSyncTimeout_forceSyncTrue() {
```

Comment:
Add a test which checks a second setPushIdentifier request which matches the first does dispatch an Edge request if outside the timeout window. You do have a test above with pushForceSync = true, but that is comparing the stored and new token values where the last sync timestamp is nil.